test/SlowTests/Tests/Faceted/FacetsWithParameters.cs

```diff
@@ -0,0 +1,423 @@
+                    };
+
+                    var query = session.Query<Foo, FooIndex>()
+                        .Where(x => x.DateIn != null && x.Age < 90)
+                        .AggregateBy(facets);
+
+                    Assert.Equal("from index 'FooIndex' where (DateIn != $p0 and Age < $p1) select " +
+                                 "facet(DateIn < $p2, DateIn >= $p3 and DateIn < $p4, DateIn >= $p5)"
+                                , query.ToString());
+
+                    var facetResult = query.Execute();
+
+                    AssertResults(facetResult, "DateIn", now);
+
+                }
+            }
+        }
+
+        [Fact]
+        public void FacetShouldUseParameters_WithIFacetBuilder()
+        {
+            using (var store = GetDocumentStore())
+            {
+                new FooIndex().Execute(store);
+
+                var now = DateTime.Now;
+                StoreSampleData(store, now);
+                WaitForIndexing(store);
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session
+                        .Query<Foo, FooIndex>()
+                        .Where(x => x.DateIn != null && x.Age < 90)
+                        .AggregateBy(builder => builder
+                            .ByRanges(
+                                c => c.DateIn < now - TimeSpan.FromDays(365),
+                                c => c.DateIn >= now - TimeSpan.FromDays(365) && c.DateIn < now - TimeSpan.FromDays(180),
+                                c => c.DateIn >= now - TimeSpan.FromDays(180)));
+
+
+                    Assert.Equal("from index 'FooIndex' where (DateIn != $p0 and Age < $p1) select " +
+                                 "facet(DateIn < $p2, DateIn >= $p3 and DateIn < $p4, DateIn >= $p5)"
+                                , query.ToString());
+
+                    var facetResult = query.Execute();
+
+                    AssertResults(facetResult, "DateIn", now);
+                }
+            }
+        }
+
+        [Fact]
+        public void TypedRangeFacetWithUntypedRangeFacetList_ShouldNotUseParameters()
+        {
+            using (var store = GetDocumentStore())
+            {
+                new FooIndex().Execute(store);
+
+                var now = DateTime.Now;
+                var oneYearAgo = DateTime.SpecifyKind(now - TimeSpan.FromDays(365), DateTimeKind.Unspecified);
+                var sixMonthsAgo = DateTime.SpecifyKind(now - TimeSpan.FromDays(180), DateTimeKind.Unspecified);
+
+                StoreSampleData(store, now);
+                WaitForIndexing(store);
+
+                using (var session = store.OpenSession())
+                {
+                    var facets = new List<RangeFacet>
+                    {
+                        //this facet will not use query-parameters when excecuting a query, due to the
+                        //implicit conversion that exists between RangeFacet<Foo> and RangeFacet
+
```

Comment:
Why does this matter?

Reply:
This is a case that we cannot support without doing breaking changes. The `RangeFacet<T>` contains an implicit conversion to `RangeFacet`. We should leave it like that.